release/src/main/scripts/download_github_actions_artifacts.py

```diff
@@ -170,7 +170,7 @@ def get_last_run_id(
       f"GCS location corresponding to artifacts built in this run: "
       f"gs://beam-wheels-staging/{release_branch}/{release_commit}-{last_run_id}/"
   )
-  return safe_get(last_run, "id")
+  return last_run_id
```

Comment:
Here is some related discussion from previous PR which introduces gh-actions:
https://github.com/apache/beam/pull/11877#issuecomment-641886596

The idea of building source dist / wheels was that it could be a good check for python related changes on PR (path filtering). I can imagine that some change could slip in which would cause problems with building wheels later on. Check on push to master and schedule also helps to find some potential bugs.
Scheduled run on master was inspired by Apache Airflow workflow: https://github.com/apache/airflow/blob/master/.github/workflows/ci.yml#L503
 What do you think about this approach?

Reply:
My concerns are mostly around quota. If actions quota is not a concern, then it's fine. 

If it is easy to configure, i would skip "Upload to GCS" steps for merges to master.

Reply:
> My concerns are mostly around quota. If actions quota is not a concern, then it's fine.

I suppose quota for actions should not be a problem ([docs](https://docs.github.com/en/actions/getting-started-with-github-actions/about-github-actions#usage-limits)). AFAIK apache has Enteprise plan.

> If it is easy to configure, i would skip "Upload to GCS" steps for merges to master.

Why would you like to skip it?

Reply:
To save GCS quota, since they are created on each commit

Reply:
I checked the lifecycle config for the bucket with wheels, looks like we delete them after 1yr. 
 
```
gsutil lifecycle get  gs://beam-wheels-staging/

{"rule": [{"action": {"type": "Delete"}, "condition": {"age": 365, "isLive": true}}, {"action": {"type": "Delete"}, "condition": {"age": 90, "isLive": false}}]}
```

Reply:
I checked what is the quota for operations on GCS and it seems huge:

> There is no limit to writes across multiple objects, which includes uploading, updating, and deleting objects. Buckets initially support roughly 1000 writes per second and then scale as needed.

source: https://cloud.google.com/storage/quotas#objects

What do you think about this lifecycle bucket settings? I checked and size of the files on the bucket is approximately 42GB, github actions are enabled for 1 month now. Rounding this numbers up, after one year it storage occupancy will be ~600GB and remains more or less constant.

Reply:
will probably be ok. if it's extra complexity to skip that step I wouldn't about it worry about it for now.